src/App.tsx

```diff
@@ -1,24 +1,305 @@
-/* eslint-disable max-len */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
-import { UserWarning } from './UserWarning';
-
-const USER_ID = 0;
+/* eslint-disable no-param-reassign */
+import React, {
+  useContext,
+  useEffect,
+  useMemo,
+  useState,
+} from 'react';
+import {
+  createTodo,
+  deleteTodo,
+  getTodos,
+  updateTodo,
+} from './api/todos';
+import { AuthContext } from './components/Auth/AuthContext';
+import { NewTodoField } from './components/NewTodoField/NewTodoField';
+import { TodoFilters } from './components/TodoFilters/TodoFilters';
+import { TodoList } from './components/TodoList/TodoList';
+import { Todo } from './types/Todo';
 
 export const App: React.FC = () => {
-  if (!USER_ID) {
-    return <UserWarning />;
-  }
+  const [isLoadError, setIsLoadError] = useState(false);
+  const [isUploadError, setIsUploadError] = useState(false);
+  const [isRemoveError, setIsRemoveError] = useState(false);
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [filtredTodos, setFiltredTodos] = useState<Todo[]>([]);
+  const [isEmptyTitle, setIsEmptyTitle] = useState(false);
+  const [isTodoLoading, setIsTodoLoading] = useState(false);
+  const [isUpdatedError, setIsUpdatedError] = useState(false);
+  const [deletedTodosId, setDeletedTodosId] = useState<number[]>([]);
+  const [activeTodoId, setActiveTodoId] = useState<number[]>([]);
+  const [isUpdatedTodo, setIsUpdatedTodo] = useState(false);
+
+  const user = useContext(AuthContext);
+
+  const loadTodosFromServer = async () => {
+    try {
+      if (user) {
+        const getTodosFromServer = await getTodos(user.id);
+
+        setTodos(getTodosFromServer);
+      }
+    } catch (error) {
+      setIsLoadError(true);
+    }
+  };
+
+  const uploadTodosOnServer = async (todo: Todo) => {
+    if (user) {
+      try {
+        setIsTodoLoading(true);
+        setActiveTodoId([0]);
+        setTempTodo(todo);
+
+        const newTodo = await createTodo({
+          title: todo.title,
+          userId: todo.userId,
+          completed: false,
+        });
+
+        setTodos(prevTodos => [...prevTodos, newTodo]);
+        setIsTodoLoading(false);
+        setTempTodo(null);
+        setActiveTodoId([]);
```

Comment:
it would be great to change the order here. Because now there is a second where the loading and newTodo is shown (check the video please)
https://user-images.githubusercontent.com/22824431/220374785-773ef058-4ee6-4902-a2c5-d5dff7907178.mov